tests/Sabre/Xml/ParserTest.php

```diff
@@ -0,0 +1,587 @@
+        $this->assertEquals('O.K. Dude', $actor1[0]['value'], "Failed to parse the actors name.");
+        $this->assertEquals(22, $actor1[1]['value'], "Failed to parse the actors age");
+        $this->assertEquals('male', $actor1[2]['value'], "Failed to parse the actors gender.");
+        $this->assertArrayHasKey(4, $movie1, "Failed to parse the bestseller element. Note that this element has no contents and no attributes.");
+        $this->assertEquals("{http://sabre-test-set.com/Movie}bestseller", $movie1[4]['name'], "Failed to parse the bestseller element. Note that this element has no contents and no attributes.");
+    }
+    
+    /**
+     * Test if the schema validation works. When enabled each node is validated on the fly.
+     * By calling the parse method the whole tree processed and thus we should get an exception.
+     */
+    function testParserShouldFailSchemaValidation()
+    {
+        $reader = new Reader();
+        $reader->XML($this->xmlWithInvalidNodes);
+        $this->assertTrue($reader->setSchema(__DIR__ . '/../../schema/Store.xsd'));
+        $failed = false;
+        try{
+            $reader->parse();
+        } catch (LibXMLException $e) {
+            $failed = true;
+        }
+        $this->assertTrue($failed, 'The parser should have failed. The element "name3" is invalid.');
+    }
+
+    /**
+     * Ensure that the validation works for valid documents
+     */
+    function testParserShouldPassSchemaValidation()
+    {
+        $reader = new Reader();
+        $reader->XML($this->xmlValidDocument);
+        $this->assertTrue($reader->setSchema(__DIR__ . '/../../schema/Store.xsd'));
+        $failed = false;
+        try{
+            $reader->parse();
+        } catch (LibXMLException $e) {
+            echo $e->getMessage() . PHP_EOL;
+            $failed = true;
+        }
+        $this->assertFalse($failed, 'The parser should not have failed. The given XML is valid.');
+    }
+
+    /**
+     * Test if the parser can handle missing start tags.
+     */
+    function testParserShouldHandleMissingStartTag()
+    {
+        // Not supported on HHVM. Not registering the tick function will lead to an eternal loop on HHVM if the test fails.
+        // But at least the test will run as long as they succeed. That's better than nothing or an exception.
+        if (function_exists('register_tick_function')) {
+            // Register tick function
+            register_tick_function(function() {
+                throw new \LogicException("Test failed. The reader seems to be trapped in a eternal loop. Failed to recognize the missing start tag...");
+            });
+        }
+
+
+        // Parse elements, use tick counter to break out of eternal loop and throw an exception
+        declare (ticks = 10000000); // Don't use low values, apparently phpunit is also using ticks and will collide with low values)
```

Comment:
I wouldn't say: get rid of the tests, but rather just get rid of the code that can abort the test if it took too long. I'm ok with the unittests getting stuck in a loop if the code is incorrect, as long as they complete when the test _is_ good.

I just want to simplify.